### ISSUE TITLE
Social Icons block can't be operated with the keyboard

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#60120]
### Description

Discovered while working on https://github.com/WordPress/gutenberg/issues/60044 / https://github.com/WordPress/gutenberg/pull/60047

After inserting a first social link inside the Social Icons block, I couldn't find a way to add _more_ links when using the keyboard.

Also, focus indication is some of the editing steps is missing.

### Step-by-step reproduction instructions

- Edit a post, add a Social Icons block.
- At this point, focus is on the empty parent block. Screenshot:

![Screenshot 2024-03-22 at 16 02 46](https://github.com/WordPress/gutenberg/assets/1682452/e02b0a87-9de6-488a-98f6-70df9b9c4001)

- Press tab and observe focus moves to the 'Add block' plus icon button. Screenshot:

![Screenshot 2024-03-22 at 16 05 51](https://github.com/WordPress/gutenberg/assets/1682452/a438f2d2-2afd-4a1e-a5e4-5e02954393e0)

- Observe the focus indication is missing.
- Press Enter and a popover to add social links opens.
- Select one of the social links and press Enter.
- The first social link is added. Screenshot:

![Screenshot 2024-03-22 at 16 03 24](https://github.com/WordPress/gutenberg/assets/1682452/fa3b28ba-c2ec-4b49-b5fa-e7a3af5aafe6)

- Observe the focus indication is missing.
- At this point, I couldn't find a way to move focus again to the 'Add block' plus icon button, Tab or Arrows key do different things.
- The process requires to move focus to the _parent_ block first. This isn't easy with the keyboard but can be done.
- From there:
  - Tab moves focus to the Settings panel. This is the expected behavior.
  - Arrow keys move focus to the inner blocks, which is the first social link.
  - Couldn't find any way to focus the 'Add block' plus icon button.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
The same issue occurs when adding Social links in a Navigation block. I couldn't find a way to mote keyboard focus to the 'plus' icon after adding a first social link:

![Screenshot 2024-03-25 at 10 03 02](https://github.com/WordPress/gutenberg/assets/1682452/756de9af-0f91-4f1d-b1d1-9de772ccfa5a)

**Comment by Mamaduka** [comment]
@afercia, the #59303 should improve situation a bit by making insert before and after shortcuts available for the inner blocks.
